diff --git a/dist/build/templates/app-page-runtime.js b/dist/build/templates/app-page-runtime.js
index f1ab03f292340fe08b563990110ac0835d3415fb..30523e38b89907c80e9cca1373574e8eff76ee82 100644
--- a/dist/build/templates/app-page-runtime.js
+++ b/dist/build/templates/app-page-runtime.js
@@ -456,7 +456,8 @@ function createAppPageEntrypoint({ tree, page, pathname, require: __next_app_req
         };
         try {
             const varyHeader = routeModule.getVaryHeader(resolvedPathname, interceptionRoutePatterns);
-            res.setHeader('Vary', varyHeader);
+            const existingVary = res.getHeader('Vary');
+            res.setHeader('Vary', existingVary ? `${existingVary}, ${varyHeader}` : varyHeader);
             let parentSpan;
             const invokeRouteModule = async (span, context)=>{
                 const nextReq = new _node.NodeNextRequest(req);
diff --git a/dist/esm/build/templates/app-page-runtime.js b/dist/esm/build/templates/app-page-runtime.js
index 1970588724b6e4880f27a8be60ca32ae6437bc28..95e6b632c240e6bcf3138fa8ccf911319ff7773b 100644
--- a/dist/esm/build/templates/app-page-runtime.js
+++ b/dist/esm/build/templates/app-page-runtime.js
@@ -471,7 +471,8 @@ export function createAppPageEntrypoint({ tree, page, pathname, require: __next_
         };
         try {
             const varyHeader = routeModule.getVaryHeader(resolvedPathname, interceptionRoutePatterns);
-            res.setHeader('Vary', varyHeader);
+            const existingVary = res.getHeader('Vary');
+            res.setHeader('Vary', existingVary ? `${existingVary}, ${varyHeader}` : varyHeader);
             let parentSpan;
             const invokeRouteModule = async (span, context)=>{
                 const nextReq = new NodeNextRequest(req);
